azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -617,4 +622,41 @@ public static void deleteSymlinkedFile(final Path symlinkedFilePath)
       throw new ExecutorManagerException(e);
     }
   }
+
+  /**
+   * Shutdown the Container. This shuts down the ExecutorService which runs the flow execution
+   * as well as JettyServer.
+   */
+  private void shutdown() {
+    logger.info("Shutting down the pod");
+    while (!this.flowFuture.isDone()) {
+      // This should not happen immediately as submitFlowRunner is a blocking call.
+      try {
+        Thread.sleep(100);
```

Comment:
Why sleep as future.get is blocking call?

Reply:
It is a good practice to add a sleep otherwise due to some bug, we may end up shutting down before flow finishes.
This is more like a sanity check. In my test runs, this code never exercised.